Sometimes, WKTR installation does not succeed in the Simulator

pvollan opened the pull request:
#### 99fde22c848d06a9d4524d99e42554c6f7f9a2ed
<pre>
Sometimes, WKTR installation does not succeed in the Simulator
<a href="https://bugs.webkit.org/show_bug.cgi?id=281875">https://bugs.webkit.org/show_bug.cgi?id=281875</a>
<a href="https://rdar.apple.com/138351968">rdar://138351968</a>

Reviewed by Elliott Williams.

This is caused by a daemon involved in the installation being suspended. This patch works around this
issue by creating a small utility that sends a XPC message to the daemon in order to resume execution.
This issue is flaky, so it is hard to confirm the fix is sufficient, but local testing looks good.

* Tools/Scripts/webkitpy/xcode/simulated_device.py:
(SimulatedDevice.install_app):
* Tools/WebKitTestRunner/Configurations/WebKitEligibilityUtil.xcconfig: Added.
* Tools/WebKitTestRunner/WebKitEligibilityUtil/main.cpp: Added.
(main):
* Tools/WebKitTestRunner/WebKitTestRunner.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/285637@main">https://commits.webkit.org/285637@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/948e8f98eb0a19a9a92a572774050190f9c7df80

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/73314 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/52743 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/26122 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/77546 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/24552 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/75429 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/60549 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/526 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/57594 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/16061 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/76381 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/47629 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/63081 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/38013 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/72822 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/44268 "Passed tests") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/20562 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-masking/clip-path/animations/clip-path-animation-font-size-mixed-change.html (failure)") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/22881 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/66119 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/20915 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/79196 "Built successfully") | 
| | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/128/builds/619 "Build was cancelled. Recent messages:OS: Sequoia (15.0.1), Xcode: 16.0; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit (cancelled)") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/154 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/66026 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/771 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/63092 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/65303 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/9150 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/7315 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/11294 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/593 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/623 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/607 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/625 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

bls1999 commented on Tools/Scripts/webkitpy/xcode/simulated_device.py:
```diff
@@ -643,6 +645,11 @@ def _tear_down(self, timeout=10.0):
     def install_app(self, app_path, env=None):
         # Even after carousel is running, it takes a few seconds for watchOS to allow installs.
         for i in range(self.NUM_INSTALL_RETRIES):
+            # FIXME: remove this workaround after rdar://129789675 has been resolved.
```
Is there a `xcrun simctl` command we can run here instead of sending a XPC message?

pvollan replied:
That is a good point! It would be better if we could just send the XPC message via the command line or script, but I have not found a way to do that yet.

Thanks for reviewing!

pvollan replied:
So far, I have not found a way to do this via the command line or via a script.

bls1999 reviewed (approved):
[unofficial] r=me. Thanks for doing this!

emw-apple reviewed (approved):
LGTM

pvollan committed:
Sometimes, WKTR installation does not succeed in the Simulator
https://bugs.webkit.org/show_bug.cgi?id=281875
rdar://138351968

Reviewed by Elliott Williams.

This is caused by a daemon involved in the installation being suspended. This patch works around this
issue by creating a small utility that sends a XPC message to the daemon in order to resume execution.
This issue is flaky, so it is hard to confirm the fix is sufficient, but local testing looks good.

* Tools/Scripts/webkitpy/xcode/simulated_device.py:
(SimulatedDevice.install_app):
* Tools/WebKitTestRunner/Configurations/WebKitEligibilityUtil.xcconfig: Added.
* Tools/WebKitTestRunner/WebKitEligibilityUtil/main.cpp: Added.
(main):
* Tools/WebKitTestRunner/WebKitTestRunner.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/285637@main
